FieldOpt/Optimization/objective/NPV.h

```diff
@@ -0,0 +1,57 @@
+//
+// Created by bragesk on 11.09.18.
+//
+
+#ifndef FIELDOPT_NPV_H
+#define FIELDOPT_NPV_H
+
+#include "objective.h"
+#include "Settings/model.h"
+#include "Simulation/results/results.h"
+
+namespace Optimization {
+    namespace Objective {
+
+        class NPV : public Objective
+        {
+        public:
+/*!
+ * \brief NPV
+ * \param settings The Settings object from which to create the objective.
+ * \param results The Results object from which to get property values.
+ */
+            NPV(Settings::Optimizer *settings,
+                Simulation::Results::Results *results);
+
+            double value() const;
+
+        private:
+/*!
+ * \brief The Component class is used for internal representation of the components of
+ * NPV.
+ */
+            class Component {
```

Comment:
I recommend using `struct` instead of `class` for internal data structures.
They are essentially the same, the only relevant difference being that all members of a struct are `public` by default.